workshops/views.py

```diff
@@ -178,6 +178,16 @@ class SiteUpdate(LoginRequiredMixin, UpdateViewContext):
     slug_url_kwarg = 'site_domain'
     template_name = 'workshops/generic_form.html'
 
+
+@login_required
+def site_delete(request, site_domain):
+    """Delete specific site."""
+    site = get_object_or_404(Site, domain=site_domain)
+    site.delete()
+
+    messages.success(request, 'Site was deleted successfully.')
```

Comment:
On Thu, Jun 11, 2015 at 07:35:16AM -0700, Piotr Banaszkiewicz wrote:

> > sounds like a lot of state to store in the expectation that you're
> > going to get a particular future request
> 
> With `return redirect()` we will get it.

Assuming that the client bothers to follow the redirect.  And what
happens if the client has simultaneous requests to two view that both
consume these messages?  You might have an event listing that shows
the “person created” message intended for the parallel person listing.
That's likely a low-probability event, but I don't like racy state ;).
And passing the message(s) as query parameters means you cover both
the racy-request case and the “client never bothered to follow the
redirect” case.